[project]
name = "ephemeral-pulumi-deploy"
version = "0.0.5"
description = "Library for facilitating easy deployment of ephemeral pulumi stacks"
authors = [
    {name = "Eli Fine"},

]
readme = "README.md"
requires-python = ">=3.12.7"
classifiers = [
    "Programming Language :: Python :: 3",

    "Programming Language :: Python :: 3.12",

    "Programming Language :: Python :: 3.13",

]
dependencies = [

    "pulumi>=3.190.0",
    "pulumi-aws>=7.5.0",
    "pulumi-aws-native>=1.32.0",
    "boto3>=1.40.9",

]


[dependency-groups]
dev = [
    # Specific to this repository

    "boto3-stubs[all]>=1.40.9",
    "pytest-mock>=3.14",

    # Managed by upstream template
    "pyright>=1.1.403",
    "pytest>=8.4.1",
    "pytest-cov>=6.2.1",
    "pytest-randomly>=3.16.0",


]

[tool.setuptools]
license-files = [] # kludge until this bug is fixed https://github.com/pypa/setuptools/issues/4759

[tool.uv]
package = true

[[tool.uv.index]]
name = "pypi"
url = "https://pypi.org/simple/"
publish-url = "https://upload.pypi.org/legacy/"

[[tool.uv.index]]
name = "testpypi"
url = "https://test.pypi.org/simple/"
publish-url = "https://test.pypi.org/legacy/"
